### PR TITLE
ACHECKER-5: Removing Deprecated Call-By-Reference PHP Codes

### PR DESCRIPTION
**Pull request by VictorAlagwu**
Removing Deprecated Call-By-Reference PHP Codes
Still In Progress

https://issues.fluidproject.org/projects/ACHECKER/issues/ACHECKER-5

### CODE REVIEW

**Comment by cindyli** [comment]
Seems this class is still using the old constructors: include/lib/tfpdf/font/unifont/ttfonts.php

**Comment by cindyli** [comment]
Line 820 and 1006 in [include/securimage/WavFile.php](https://github.com/VictorAlagwu/AChecker/blob/ACHECKER-5/include/securimage/WavFile.php) still use deprecated passing by reference (&$). Please search thru the code base to address all. Thanks.

**Comment by cindyli** [comment]
I wonder why the recently upgraded secureImage library uses deprecated passing by reference. Doesn't it claim it supports PHP7?

**Comment by VictorAlagwu** [comment]
Yes, it does support PHP 7

**Comment by cindyli** [comment]
Testing this branch with PHP 7 shows the theme is not applied:
screen shot 2018-06-27 at 10 44 26 am

Also, a PHP error is reported: `Invalid argument supplied for foreach() in ../AChecker/themes/default/include/header.tmpl.php on line 120`

I suspect these issues are caused by the fact that the savant2 template system doesn't support PHP 7. What do you think?

**Comment by VictorAlagwu** [comment]
Yes, it is as a result of Savant2 not supporting PHP 7, I would include a little patch for it to work,for the main time, till I complete the task on changing the template engine

File: include/classes/BasicChecks.class.php [hunk 1]
```diff
@@ -160,7 +160,7 @@ public static function getNumOfTagRecursiveInChildren($e, $tag)
 	*/
 	public static function hasDuplicateAttribute($e, $attr, $id_array)
 	{
-		global $has_duplicate_attribute, $id_array;
+		global $has_duplicate_attribute;
```
Comment (cindyli) [inline include/classes/BasicChecks.class.php]: I wonder why `global $id_array` is removed? I believe this declaration is to make sure all ids from recursive function calls at line 179 are tracked in it. Without this declaration, each recursive call will start a new local `$id_array` only with ids captured in this single call.

File: include/classes/BasicChecks.class.php [hunk 2]
```diff
@@ -510,7 +510,7 @@ public static function get_luminosity_contrast_ratio($color1, $color2)
 	*/
 	public static function has_duplicate_attribute($e, $attr, $id_array)
 	{
-		global $has_duplicate_attribute, $id_array;
+		global $has_duplicate_attribute;
```
Comment (cindyli) [inline include/classes/BasicChecks.class.php]: Same comment about `global $id_array` as above.

File: include/classes/BasicChecks.class.php (hunk 1 above)
Comment (VictorAlagwu) [reply include/classes/BasicChecks.class.php]: I notice including that, do result to this error message being shown in both PHP 5.6 and 7
errorbasiccheck

**Pull request merged**